src/azure-cli/azure/cli/command_modules/resource/_utils.py

```diff
@@ -5,47 +5,61 @@
 
 import re
 
+_management_group_pattern = (
+    r"^\/?providers\/Microsoft.Management\/managementGroups\/(?P<management_group_id>[\w\d_\.\(\)-]+)"
+)
 _subscription_pattern = r"^\/?subscriptions\/(?P<subscription_id>[a-f0-9-]+)"
 _resource_group_pattern = r"^\/resourceGroups\/(?P<resource_group_name>[-\w\._\(\)]+)"
 _relative_resource_id_pattern = r"^\/providers/(?P<relative_resource_id>.+$)"
 
 
-def parse_resource_id(resource_id):
+def split_resource_id(resource_id):
```

Comment:
It is suggested to add some comments and examples to this method to help others understand its logic and function more clearly when using it.

Reply:
Thanks for the suggestion, added examples.

Reply:
Not sure if you can leverage existing method "parse_resource_id" in azure.mgmt.core as it seems very similar to yours

---
In reply to: [462710200](https://github.com/Azure/azure-cli/pull/14568#discussion_r462710200) [](ancestors = 462710200)

Reply:
It is similar, but it may not be suitable to use it here, because what the `parse_resource_id` method returns is a dictionary with unpredictable number of key/value pairs depending on how many child resource types a resource ID has. If using it, I have to process the dictionary to construct the relative resource ID, which appears to be a harder task.